WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewViewModel.swift

```diff
@@ -212,6 +213,58 @@ private extension ProductDetailPreviewViewModel {
     }
 }
 
+// MARK: - Site settings
+//
+private extension ProductDetailPreviewViewModel {
+    func fetchSettingsIfNeeded() async {
+        guard weightUnit == nil || dimensionUnit == nil else {
+            return
+        }
+
+        await withTaskGroup(of: Void.self) { group in
+            group.addTask {
+                try? await self.fetchGeneralSettings()
+            }
+            group.addTask {
+                try? await self.fetchProductSiteSettings()
+            }
+        }
+
+        currency = ServiceLocator.currencySettings.symbol(from: ServiceLocator.currencySettings.currencyCode)
+        currencyFormatter = CurrencyFormatter(currencySettings: ServiceLocator.currencySettings)
+        weightUnit = ServiceLocator.shippingSettingsService.weightUnit
+        dimensionUnit = ServiceLocator.shippingSettingsService.dimensionUnit
```

Comment:
💡 For testability, we should inject `ShippingSettingsService` to the view model and check its unit after fetching succeeds. I'm not sure if we can do the same with `CurrencySettings` though, could you check?

Reply:
Sorry, I am not sure I understand this. Do you recommend mocking the `ShippingSettingsService`? Should we also update core data with a shipping unit? Sorry, I can't understand what we should mock and what we should test. 

- My current approach is to test that we send a request to synchronize the site and product settings when the weight or dimension unit is nil. 
- As there is no return value from the synchronize actions, I think it is enough to test that we send a request to synchronize the settings when needed. 

Please correct me if I am wrong.

Reply:
Since we are getting the value from `ServiceLocator.shippingSettingsService.weightUnit`, I think it'd be beneficial to also make sure that the synced unit is saved correctly to storage. But now that I think of it, we can't really inject the same storage to the sync call, and that's more like integration test, not unit test. Please feel free to skip this comment then.

Reply:
Thanks for sharing the details!